reconcile/test/test_vpc_peerings_validator.py

```diff
@@ -101,3 +110,65 @@ def test_validate_no_public_to_public_peerings_valid(
 ):
     query_data_p2p.clusters[0].peering.connections[0].cluster.spec.private = True  # type: ignore[index,union-attr]
     assert validate_no_public_to_public_peerings(query_data_p2p) is True
+
+
+def test_validate_validate_no_cidr_overlap():
+    test_dict = {
+        "cidr_block": "10.25.0.0/16",
+        "providers": [
+            {
+                "provider": "account-vpc",
+                "vpc_peering": {
+                    "vpc_name": "vpc-name-1",
+                    "cidr_block": "10.18.0.0/18",
+                },
+            },
+            {
+                "provider": "account-vpc",
+                "vpc_peering": {
+                    "vpc_name": "vpc-name-2",
+                    "cidr_block": "10.18.0.0/18",
+                },
+            },
+        ],
+    }
```

Comment:
The main idea is to treat cluster-self vpc as a special provider, so all the peerings_enteries are in the same format. The example test data can be changed form
```
   {
        "cidr_block": "10.25.0.0/16",
        "providers": [
            {
                "provider": "account-vpc",
                "vpc_peering": {
                    "vpc_name": "vpc-name-1",
                    "cidr_block": "10.18.0.0/18",
                },
            },
            {
                "provider": "account-vpc",
                "vpc_peering": {
                    "vpc_name": "vpc-name-2",
                    "cidr_block": "10.18.0.0/18",
                },
            },
        ],
    }
```
to 
```
   [
            {
                "provider": "cluster-self-vpc",
                "vpc_name": "cluster-name",
                "cidr_block": "10.25.0.0/16",
            },
            {
                "provider": "account-vpc",
                "vpc_name": "vpc-name-1",
                "cidr_block": "10.18.0.0/18",
            },
            {
                "provider": "account-vpc",
                "vpc_name": "vpc-name-2",
                "cidr_block": "10.18.0.0/18",
            },
    ]
```
Notice I removed "vpc_peering" to make the array item flat.